File: app/src/main/java/ai/rotor/rotorvehicle/ctl/ArduinoListener.java
package ai.rotor.rotorvehicle.ctl;

import android.hardware.usb.UsbDevice;

public interface ArduinoListener {
    void onArduinoAttached(UsbDevice mLastArduinoAttached);

    void onArduinoDetached();

    void onArduinoOpened();

    void onUsbPermissionDenied();

    void onArduinoMessage(String message);
}

File: app/src/main/java/ai/rotor/rotorvehicle/ctl/ArduinoListener.kt
package ai.rotor.rotorvehicle.ctl

import android.hardware.usb.UsbDevice

interface ArduinoListener {
    fun onArduinoAttached(mLastArduinoAttached: UsbDevice)

    fun onArduinoDetached()

    fun onArduinoOpened()

    fun onUsbPermissionDenied()

    fun onArduinoMessage(message: String)
}
